modules/ocf_desktop/manifests/xsession.pp

```diff
@@ -43,7 +47,8 @@
     '/opt/share/xsession/icons':
       source  => 'puppet:///modules/ocf_desktop/xsession/icons/',
       recurse => true,
-      purge   => true;
+      purge   => true,
+      before  => File['/usr/local/bin/paper-genmon'];
```

Comment:
imo it's usually better for things to specify their dependencies, rather than other things specifiying them as reverse dependencies

i.e. i'd write `require => File['/opt/share/xsession/icons']` on the script instead of this